Tests/FoundationEssentialsTests/FileManager/FileManagerTests.swift

```diff
@@ -1046,6 +1046,70 @@ private struct FileManagerTests {
         }
     }
 
+    #if !os(Windows) && !os(WASI) && !os(OpenBSD) && !canImport(Android)
+    @Test func extendedAttributesDoNotFollowSymlinksWhenSetting() async throws {
+        let xattrKey = FileAttributeKey("NSFileExtendedAttributes")
+        #if canImport(Darwin)
+        let attrName = "com.swiftfoundation.symlinktest"
+        #elseif os(Linux)
+        // Linux requires the user.* namespace prefix for regular files
+        let attrName = "user.swiftfoundation.symlinktest"
+        #else
+        let attrName = "swiftfoundation.symlinktest"
+        #endif
+        let attrValue = Data([0xAA, 0xBB, 0xCC])
+
+        try await FilePlayground {
+            File("target", contents: Data("payload".utf8))
+            SymbolicLink("link", destination: "target")
+        }.test { fileManager in
+            // Attempt to set xattrs on the symlink; if unsupported, bail quietly.
+            do {
+                try fileManager.setAttributes(
+                    [xattrKey: [attrName: attrValue]], ofItemAtPath: "link")
+            } catch let error as CocoaError {
+                if error.code == .featureUnsupported {
+                    return
+                }
+                if let posix = error.underlying as? POSIXError,
+                    posix.code == .ENOTSUP || posix.code == .EOPNOTSUPP
+                {
+                    return
+                }
+                throw error
+            }
+
+            guard
+                let linkAttrs = try? fileManager.attributesOfItem(atPath: "link"),
+                let targetAttrs = try? fileManager.attributesOfItem(atPath: "target")
+            else { return }
+
+            let linkXattrs = linkAttrs[xattrKey] as? [String: Data]
+            let targetXattrs = targetAttrs[xattrKey] as? [String: Data]
+
+            // If neither reports xattrs, treat as unsupported on this FS/runtime.
+            if linkXattrs == nil && targetXattrs == nil {
+                return
```

Comment:
I think in this case and the case above when calling `attributesOfItem` we likely don't want to just `return` if the call fails / the value is `nil`. If we successfully set the extended attributes above (i.e. `setAttributes` didn't throw) we should expect that we can then read that extended attribute since any system that doesn't support this operation would have already exited the test. Do you see another reason to allow for the test to pass when writing succeeds but reading fails? Or instead, should we just call the API with `try` and `#require` the attribute values so that if we succeed writing the attributes but fail reading them will cause the test to fail and report the regression